test/test-exception.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+// Testcase to produce NodeReport on uncaught exception
+if (process.argv[2] === 'child') {
+  require('../').setEvents("exception");
+
+  function myException(request, response) {
+    throw new UserException('*** exception.js: testcase exception thrown from myException()');
+  }
+
+  function UserException(message) {
+    this.message = message;
+    this.name = "UserException";
+  }
+
+  myException(); 
+} else {
+  const tap = require('tap');
+  const common = require('./common.js');
+  const fs = require('fs');
+  const spawn = require('child_process').spawn;
+
+  var child = spawn(process.execPath, [__filename, 'child']);
+  child.on('exit', function(code, signal) {
+    tap.plan(3);
+    tap.notEqual(code, 0, 'Process should not exit cleanly');
```

Comment:
Maybe more of a "how Sam thinks it should work" :-). I'd do `tap.notEqual(signal || code, 0, ...` <-- note carefully that signal is before code, because `0` is a valid code and is false. Might have to change this back for now.